src/main/scala/com/fulcrumgenomics/commons/io/Io.scala

```diff
@@ -49,8 +50,13 @@ trait IoUtil {
 
   /** Creates a new InputStream to read from the supplied path. */
   def toInputStream(path: Path) : InputStream = {
-    val stream =  if (Files.isSameFile(path, Io.StdIn)) System.in else Files.newInputStream(path)
-    new BufferedInputStream(stream, bufferSize)
+    if (Files.isSameFile(path, Io.StdIn)) System.in
+    else {
+      val stream = Files.newInputStream(path)
+      val attrs  = Files.readAttributes(path, classOf[BasicFileAttributes])
+      if (!attrs.isRegularFile && !attrs.isSymbolicLink && attrs.isOther) stream
```

Comment:
Can you add some comments here so we don't re-break this?  I'm curious why buffering those streams causes a problem, since it's also possible client code will attempt to wrap the input stream in a buffer.